queue_services/entity-emailer/src/entity_emailer/email_processors/restoration_notification.py

```diff
@@ -0,0 +1,177 @@
+                'effectiveDateTime': effective_date if effective_date != filing_date_time else '',
+                'filingIdentifier': str(filing.id),
+                'businessNumber': business_data.tax_id if business_data and business_data.tax_id else ''
+            },
+            headers=headers
+        )
+        if receipt.status_code != HTTPStatus.CREATED:
+            logger.error('Failed to get receipt pdf for filing: %s', filing.id)
+        else:
+            receipt_encoded = base64.b64encode(receipt.content)
+            pdfs.append(
+                {
+                    'fileName': 'Receipt.pdf',
+                    'fileBytes': receipt_encoded.decode('utf-8'),
+                    'fileUrl': '',
+                    'attachOrder': attach_order
+                }
+            )
+            attach_order += 1
+    elif status == Filing.Status.COMPLETED.value:
+        filing_pdf = requests.get(
+            f'{current_app.config.get("LEGAL_API_URL")}/businesses/{business["identifier"]}/filings/{filing.id}',
+            headers=headers
+        )
+        if filing_pdf.status_code != HTTPStatus.OK:
+            logger.error('Failed to get pdf for filing: %s', filing.id)
+        else:
+            filing_pdf_encoded = base64.b64encode(filing_pdf.content)
+            pdfs.append(
+                {
+                    'fileName': 'Statement of Registration.pdf',
+                    'fileBytes': filing_pdf_encoded.decode('utf-8'),
+                    'fileUrl': '',
+                    'attachOrder': attach_order
+                }
+            )
+            attach_order += 1
+
+    return pdfs
+
+
+def process(email_info: dict, token: str) -> dict:  # pylint: disable=too-many-locals, , too-many-branches
+    """Build the email for Restoration notification."""
+    logger.debug('registration_notification: %s', email_info)
+    # get template and fill in parts
+    filing_type, status = email_info['type'], email_info['option']
+    # get template vars from filing
+    filing, business, leg_tmz_filing_date, leg_tmz_effective_date = get_filing_info(email_info['filingId'])
+    filing_name = filing.filing_type[0].upper() + ' '.join(re.findall('[a-zA-Z][^A-Z]*', filing.filing_type[1:]))
+    business = business if business else filing.json['filing']['business']
+    identifier = business.get('identifier')
+    filing_business = filing.json['filing']['business']
+    corp_type = CorpType.find_by_id(filing_business.get('legalType'))
+    restoration_type = filing.json['filing']['restoration']['type']
+
+    # create the jinja environment
+    jinja_env = Environment(
+        loader=FileSystemLoader(current_app.config.get("TEMPLATE_PATH")),
+        autoescape=True
+    )
+    # look for a template in this format RES-fullRestoration-PAID.html
+    # if the template doesn't exists use RES-PAID.html
+    template = jinja_env.select_template([
+        f'RES-{restoration_type}-{status}.jinja2',
+        f'RES-{status}.jinja2'
+    ])
```

Comment:
Rather than calling a single template, here we supply a list of templates.  `select_template()` checks for the templates in list order.  If the template doesn't exist, it trys the next.  This pattern makes it very easy to create an exception to the default template without cluttering it up with an `if-then`.